backend-app/controllers/users_controllers/user_controller.ts

```diff
@@ -1,72 +1,120 @@
+} from 'tsoa';
+import User from '@root/models/user/user_model';
 
-export const getMe = (req: IReq, res: IRes) => {
-    // return data of the current user
-    res.status(200).json({
-        user: req.user,
-    });
-};
-
-export const deleteMe = async (req: IReq, res: IRes, next: INext) => {
-    try {
-        await User.findByIdAndUpdate(req.user._id, {
-            deleted: true,
-            deletedAt: Date.now(),
-            deletedBy: req.user._id,
-        });
-
-        res.status(204).json({
-            data: null,
-        });
-    } catch (error) {
-        next(error);
+@Route('api/users')
+@Security('jwt')
+@Tags('User')
+export class UserController extends Controller {
+    @Response(401, 'Unauthorized')
+    @SuccessResponse(200, 'OK')
+    @Get('me')
+    public getMe(@Request() req: IReq): Promise<any> {
+        if (!req.user) {
+            return Promise.reject(new AppError(401, 'Please log in again!'));
+        }
+        // return data of the current user
+        return Promise.resolve(req.user);
     }
-};
 
-export const updateMe = async (req: IReq, res: IRes, next: INext) => {
-    try {
-        // 1) Create error if user POSTs password data
-        if (req.body.password || req.body.passwordConfirm) {
-            throw new AppError(
-                400,
-                'This route is not for password updates. Please use /updateMyPassword'
-            );
-        }
-        // create error if user tries to update role
-        if (req.body.roles) {
-            throw new AppError(
-                400,
-                'This route is not for role updates. Please use /updateRole'
-            );
-        }
-        // 2) Filtered out unwanted fields names that are not allowed to be updated
-        const payload = {
-            name: req.body.name,
-            email: req.body.email,
-        };
-        // 3) Update user document
-        const doc = await User.findByIdAndUpdate(req.user._id, payload, {
-            new: true,
-            runValidators: true,
-        });
-        if (!doc) {
-            return next(new AppError(404, 'No document found with that id'));
+    @Response(401, 'Unauthorized')
+    @SuccessResponse(204, 'No Content')
+    @Delete('me')
+    public async deleteMe(
+        @Request() req: IReq,
+        @Res() res: TsoaResponse<204, { message: string }>
+    ): Promise<void> {
+        try {
+            await User.findByIdAndUpdate(req.user._id, {
+                deleted: true,
+                deletedAt: Date.now(),
+                deletedBy: req.user._id,
+            });
+            return res(204, { message: 'User deleted successfully' });
+        } catch (error) {
+            return Promise.reject(new AppError(400, error.message));
         }
+    }
+
+    @Response(401, 'Unauthorized')
+    @Response(
+        400,
+        `- This route is not for role updates. Please use /updateRole\n - This route is not for password updates. Please use auth/updateMyPassword`
```

Comment:
Can we use?  :
```txt
<ul> 
<li>text1<li\> 
<li>text2<li\>
<ul\>
```

Reply:
I'm not sure if it will work ! But I will test it